omniscidb/QueryEngine/Visitors/TemplateAggregationVisitor.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include "QueryEngine/CostModel/Measurements.h"
+#include "QueryEngine/RelAlgVisitor.h"
+
+#include <unordered_set>
+#include <vector>
+
+struct TemplateSample {
+  bool consider;
+  costmodel::AnalyticalTemplate templ;
+};
+
+class TemplateAggregationVisitor : public RelAlgVisitor<TemplateSample> {
+ public:
+  virtual TemplateSample visitAggregate(const hdk::ir::Aggregate* n) const {
+    bool consider = true;
+    if (visited_nodes_.find(n) != visited_nodes_.end())
+      consider = false;
+    else
+      visited_nodes_.insert(n);
+    return {consider, costmodel::AnalyticalTemplate::GroupBy};
```

Comment:
```suggestion
    bool consider = visited_nodes_.count(n);
    if (!consider) {
      visited_nodes_.insert(n);
    }
    return {consider, costmodel::AnalyticalTemplate::GroupBy};
```
And maybe move the code to some private method to avoid duplication.